File: frontend/src/types/pauta.ts
export interface Vara {
  id: string;
  codigo: string;
  nome: string;
}

export interface PautasResponse {
  varas: Vara[];
  datas_disponiveis: string[];
}

export interface ExtrairRequest {
  vara_ids: string[];
  datas: string[];
}

export interface ExtrairResponse {
  jobs_iniciados: number;
  keys: string[];
}

export interface ExtrairJobStatus {
  key: string;
  vara_id: string;
  data: string;
  status: "running" | "done" | "error";
  processos_encontrados: number;
  leads_criados: number;
  processos_com_advogado: number;
  errors: string[];
}

File: backend/app/models/pauta.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import date


class Vara(BaseModel):
    id: str
    codigo: str
    nome: str


class ExtrairRequest(BaseModel):
    vara_ids: List[str]
    datas: List[date]


class ExtrairJobStatus(BaseModel):
    key: str
    vara_id: str
    vara_nome: str = ""
    data: date
    status: str  # "running" | "done" | "error"
    processos_encontrados: int = 0
    leads_criados: int = 0
    processos_com_advogado: int = 0
    errors: List[str] = []


class ExtrairResponse(BaseModel):
    jobs_iniciados: int
    keys: List[str]


class PautasResponse(BaseModel):
    varas: List[Vara]
    datas_disponiveis: List[date]
